meson.build

```diff
@@ -1123,3 +1123,54 @@ if get_option('unit-tests-backends')
     endif
   endforeach
 endif
+
+if python.found()
+  find_rst_files = run_command(['find', docs_dir, '-name', '*.rst'])
+  rst_files = find_rst_files.stdout().strip().split('\n')
```

Comment:
This won't be updated when new files are added. The QEMU project faced this challenge already, and wrote a neat trick to solve it:

https://gitlab.com/qemu-project/qemu/-/blob/master/docs/sphinx/depfile.py

This allows you to use `depfile: ...` and have sphinx report what files it used and which should trigger further rebuilds.

Reply:
Ah, I mentioned the drawback in the commit message. I'll look into it.

Reply:
Yes, Meson isn't a fan of this run_command method... because it's better done by depfiles any time you don't explicitly need the input files as command line arguments! :)

Reply:
> Actually, the depfile (sphinx.d) is removed by nija after generating
> it as shown by a trace.  I do not know why, but touching an .rts
> file has the correct behaviour. So I suspoect it is an optimization.

--> `ninja -d list`

```
debugging modes:
[...]
  keepdepfile  don't delete depfiles after they're read by ninja
```

We use depfiles for all *.o outputs as well. Ninja reads them in after the command finishes for build nodes that declare they provide depfiles, and compiles them into a binary database (`.ninja_deps`) for, as you guessed, optimization purposes. No need to read tons of tiny files in on every startup. :)

Hence also why keeping the file around is one of the options exposed in the *debugging* modes.